doc/man1/flux-pgrep.rst

```diff
@@ -0,0 +1,98 @@
+.. flux-help-include: true
+
+==============
+flux-pgrep(1)
+==============
+
+
+SYNOPSIS
+========
+
+**flux** **pgrep** [*OPTIONS*] expression..
+
+**flux** **pkill** [*OPTIONS*] expression..
+
+DESCRIPTION
+===========
+
+*flux-pgrep* lists jobids that match a supplied expression. The
+expression may contain a pattern which matches the job name, or
+a range of jobids in the form ``jobid1..jobid2``. If both a pattern
+and jobid range are supplied then both must match.
+
+In rare cases, a pattern may appear to be a jobid range, when the
+pattern ``..`` is used and the strings on both sides area also valid
+Flux jobids. In that case, a name pattern match can be forced by
+prefixing the pattern with ``name:``, e.g. ``name:fr..123``.
+
+By default, only active jobs for the current user are considered.
+
+*flux-pkill* cancels matching jobs instead of listing them. This is
+equivalent to running *flux-pgrep* with the ``-k`` option.
+
+OPTIONS
+=======
+
+**-a**
+   Include jobs in all states, including inactive jobs.
+   This is shorthand for *--filter=pending,running,inactive*.
```

Comment:
No this is for `pkill` also, but was there something that led you to believe `-a` was for `pgrep` only?

Reply:
```
   parser.add_argument(
        "-a",
        action="store_true",
        help="Target jobs in all states"
        if PROGRAM == "flux-pgrep"
        else argparse.SUPPRESS,
    )
```

you suppress the option if its pgrep (unless I'm reading this wrong, I assume black formatted things on different lines)

Reply:
Oh, yeah, duh, that makes sense (I was thinking -A) since you can only cancel active jobs :facepalm: I like your idea of marking options `pgrep-only`

Reply:
I wonder if `--filter` should also be suppressed in `flux-pkill`? (Sorry, I hadn't looked at this portion of the code since before vacation, so thanks for catching that one!)

Reply:
I guess filter still may be useful to cancel only pending jobs for example.

Reply:
i think keeping `--filter` is a good idea for both.

Reply:
> Oh, yeah, duh, that makes sense (I was thinking -A) since you can only cancel active jobs facepalm I like your idea of marking options pgrep-only

Noticed the way pgrep(1) does it is:

```
       -signal
       --signal signal
              Defines the signal to send to each matched process.  Either the numeric or the symbolic signal name can be  used.   (pkill only.)

       -d, --delimiter delimiter
              Sets the string used to delimit each process ID in the output (by default a newline).  (pgrep only.)
```

which I sort of like.

Reply:
Yeah, agreed. I'll update the man page to use that style.

Reply:
it just occurred to me, SUPPRESS hides the option, doesn't remove it.  was that intentional?  I guess if someone did `-a` with `flux-pkill` it's not the end of the world, although we'd be trying to cancel a bunch of inactive jobs?

Reply:
This option needs to use SUPPRESS because `args.a` is used in the code when fetching jobs, otherwise Python would throw an exception there that `a` doesn't exist in the `Namespace`. We could add a check that `args.a` isn't set when `PROGRAM == "flux-pkill"` but that didn't seem worth it for now, and it occurs to me that might be useful in testing.